components/google-cloud/google_cloud_pipeline_components/container/experimental/notebooks/executor.py

```diff
@@ -0,0 +1,275 @@
+        gcp_resources=gcp_resources)
+
+  # Waits for execution to finish.
+  print('Blocking pipeline...')
+  execution_state = ''
+  execution_job_uri = ''
+  while True:
+    try:
+      execution = client_notebooks.get_execution(name=execution_fullname)
+      execution_state = getattr(execution, 'state', '')
+      print(f'execution.state is {Execution.State(execution_state).name}')
+    except Exception as e:
+      response = build_response(
+          error=f'get_execution() for blocking pipeline failed: {e}')
+      handle_error(args.fail_pipeline, response)
+      return response
+
+    # Job URI is not available when state is INITIALIZING.
+    if execution_state in _STATES_JOB_URI and not execution_job_uri:
+      execution_job_uri = getattr(execution, 'job_uri', '')
+      print(f'execution.job_uri is {execution_job_uri}')
+
+    if execution_state in _STATES_COMPLETED:
+      break
+    time.sleep(30)
+
+  # For some reason, execution.get and job.get might not return the same state
+  # so we check if we can get the error message using the AI Plaform API.
+  # It is only possible if there is job_uri available though.
+  if execution_state in _STATES_ERROR:
+    if execution_job_uri:
+      while True:
+        try:
+          custom_job = client_vertexai_jobs.get_custom_job(name=execution_job_uri)
+        except Exception as e:
+          response = build_response(error=f'get_custom_job() failed: {e}')
+          handle_error(args.fail_pipeline, response)
+          return response
+
+        custom_job_state = getattr(custom_job, 'state', None)
+        if custom_job_state in _STATES_COMPLETED:
+          break
+        time.sleep(30)
+
+      # == to `if state in _JOB_ERROR_STATES`
+      custom_job_error = getattr(custom_job, 'error', None)
+      if custom_job_error:
+        response = build_response(
+            error=f'Error {custom_job_error.code}: {custom_job_error.message}')
+        handle_error(args.fail_pipeline, (None, response))
+        return response
+
+    # The job might be successful but we need to address that the execution
+    # had a problem. The previous loop was in hope to find the error message,
+    # we didn't have any so we return the execution state as the message.
+    response = build_response(
+        error=f'Execution finished with state: {Execution.State(execution_state).name}')
+    handle_error(args.fail_pipeline, (None, response))
+    return response
+
+  return build_response(
+      state=Execution.State(execution_state).name,
+      output_notebook_file=execution.output_notebook_file,
+      gcp_resources=gcp_resources)
+
+def main():
+  def _deserialize_bool(s) -> bool:
+    from distutils.util import strtobool
+    return strtobool(s) == 1
+
+  def _serialize_str(str_value: str) -> str:
+    if not isinstance(str_value, str):
+      raise TypeError('Value "{}" has type "{}" instead of str.'.format(str(str_value), str(type(str_value))))
+    return str_value
+
+  import argparse
```

Comment:
Note that this interface is not compatible with serverless launcher, this is not a blocker for this release though in the future as you move to using serverless launcher the interface will need to be updated accordingly.